### PR TITLE
Debug menu cleanup

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Based of @sphericalice's documentation.

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

File: src/debug.c [hunk 1]
```diff
@@ -217,11 +238,11 @@ enum { //Sound
 // *******************************
 struct DebugMonData
 {
-    u16 mon_speciesId;
-    u8  mon_level;
-    u8  isShiny;
-    u16 mon_natureId;
-    u16 mon_abilityNum;
+    u16 species;
+    u8 level:7;
```
Comment (DizzyEggg) [inline src/debug.c:242]: I'd keep it as u8 in case someone decided to go over level 100.

File: src/debug.c [hunk 2]
```diff
@@ -217,11 +238,11 @@ enum { //Sound
...
+    bool8 isShiny:1;
+    u8 nature:6;
```
Comment (DizzyEggg) [inline src/debug.c:244]: This can take 5 bits I think.

File: src/debug.c [hunk 3]
```diff
@@ -2549,72 +2636,73 @@ static void DebugAction_Give_Item_SelectId(u8 taskId)
...
         StringCopyPadded(gStringVar1, gStringVar1, CHAR_SPACE, 15);
         StringExpandPlaceholders(gStringVar4, sDebugText_ItemQuantity);
-        AddTextPrinterParameterized(gTasks[taskId].data[2], 1, gStringVar4, 1, 1, 0, NULL);
+        AddTextPrinterParameterized(gTasks[taskId].tSubWindowId, DEBUG_MENU_FONT, gStringVar4, 1, 1, 0, NULL);
 
         gTasks[taskId].func = DebugAction_Give_Item_SelectQuantity;
     }
     else if (JOY_NEW(B_BUTTON))
     {
-        FreeSpriteTilesByTag(ITEM_TAG);                         //Destroy item icon
```
Comment (DizzyEggg) [inline src/debug.c:2606]: The comment can stay.

**Review by DizzyEggg — CHANGES_REQUESTED** [review]

File: src/debug.c (hunk 3 above)
Comment (AsparagusEduardo) [reply src/debug.c:2606]: How about a new `DestroyItemIcon` function? I saw this section of code 4 times, all with the same comment

**Comment by DizzyEggg** [comment]
Conflicts

**Comment by AsparagusEduardo** [comment]
Conflicts fixed

**Comment by DizzyEggg** [comment]
Conflicts again xD

**Comment by AsparagusEduardo** [comment]
Solved conflicts. Also fixed issue found in #3382.

**Review by DizzyEggg — APPROVED** [review]

**Pull request merged**